.eslintrc.js

```diff
@@ -0,0 +1,37 @@
+const os = require('os');
+
+module.exports = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 8,
+    sourceType: 'module',
+    ecmaFeatures: {
+      experimentalObjectRestSpread: true
+    }
+  },
+  rules: {
+    '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
+    indent: ['error', 2, { SwitchCase: 1 }],
+    'linebreak-style': ['error', os.EOL === '\r\n' ? 'windows' : 'unix'],
```

Comment:
I ran into a lot of problems with this.  Let me start at the beginning...

Assume the following:

1. Clone repo on a Windows machine
2. Use VS Code
3. Use the most popular ESLint extension in VS Code: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

In this case, many files will be entirely redlined as below:

![image](https://user-images.githubusercontent.com/109760379/193662994-80fe7890-3aac-4d9e-8ff9-ccf8878c2a53.png)

This isn't just an artifact of the extension.  If I then run the "lint:fix" npm script, I'll get ~140 changed files, of whose changes I can't see:
![image](https://user-images.githubusercontent.com/109760379/193663546-52bfb84f-35b1-4799-847d-0af2b82d0a47.png)

So I was looking for a solution to those 2 problems that wouldn't introduce a regression on unix.

The simplest solution is to turn off line-break checking.  That's easy to do, but obviously will have a regression in that tests are no longer performed.

If I change the setting to linebreak-style: ["error", "windows"], then everything works fine on the windows machine, but this will introduce a regression for users on unix after this change is committed. 

I tried the .gitattributes option, and I couldn't get it to work.  It might need to be present in the git repo before cloning to the Windows machine.  If that's the case, then we'd need to ask all Windows users to wipe and re-clone the repo.  Otherwise, the problems I mentioned above remain.  I'm open to revisiting if someone knows how to get this to work on an existing repo (maybe I missed something?).

In the end, the simplest fix I could find that resolved the issue without some sort of regression was switching to .js and adding an OS check.  That does rely on git doing the line ending conversion, which I tested here on the checks on commit.

Reply:
I thought that I had seen a Git command that would normalize your line endings in-place, but now that I'm looking for it, I can't find it. Perhaps it does not exist after all. You may have to re-clone to get this working with gitattributes.

TBH, I'm not sure if we need to be concerned about asking all the Windows users to wipe and reclone... The issues you found have been like that for a while, and no one else has ever reported them. So I wonder how many active Windows cloners we have out there...  For the ones we do have, they'd be no worse off than they already were.

I don't know how much this really matters.  Let's get a couple of 2nd/3rd opinions.  @mgramigna / @hossenlopp -- what are your thoughts on dynamically changing expected EOL in the lint config vs trying to leverage gitattributes (and requiring Windows users to reclone)?

Reply:
My personal opinion is slightly adverse to conditionally defining a lint rule based on the OS

Reply:
I'm going to agree. It's not hard to set git to enforce unix line endings on windows.